test/conftest.py

```diff
@@ -59,6 +60,41 @@ async def client_fixture(loop, client_session, ws_client, uuid4):
     return client
 
 
+@pytest.fixture(name="mock_command")
+def mock_command_fixture(ws_client, client, uuid4):
+    """Mock a command and response."""
+    mock_responses: List[Tuple[dict, dict]] = []
+    ack_commands: List[dict] = []
+
+    def apply_mock_command(match_command: dict, response: dict) -> List[dict]:
+        """Apply the mock command and response return value to the transport.
+
+        Return the list with correctly acknowledged commands.
+        """
+        mock_responses.append((match_command, response))
+        return ack_commands
+
+    async def set_response(message):
+        """Check the message and set the mocked response if a command matches."""
+        for match_command, response in mock_responses:
+            if all(message[key] == value for key, value in match_command.items()):
+                ack_commands.append(message)
+                received_message = {
+                    "type": "result",
+                    "messageId": uuid4,
+                    "result": response,
+                    "success": response["success"],
+                }
+                client.async_handle_message(received_message)
```

Comment:
If you really want to mock the transport, you would not call this method directly but instead have this message be set as result for the open `msg = await client.receive()` 😉

Reply:
(it's fine like this)